lib/util/connection-leasing.js

```diff
@@ -0,0 +1,26 @@
+const LeaseConnectionFromClient = (client) =>
+  async function withConnection(next) {
+    const connection = await client.acquireConnection();
+    try {
+      return await next(connection);
+    } finally {
+      await client.releaseConnection(connection);
+    }
+  };
+
+function ReuseConnection(connection) {
+  let previous = Promise.resolve();
+
+  function reuseConnection(next) {
+    const basePromise = previous.then(() => next(connection));
+    previous = basePromise.catch((err) => {});
+    return basePromise.then((x) => x);
+  }
+
+  return reuseConnection;
+}
+
+Object.assign(exports, {
```

Comment:
Why Object.assign and not module.exports = {}?

Reply:
`Object.assign(..)` tends to behave slightly better when you have circular dependencies.  But yeah: you really shouldn't have circular dependencies in the first place.  lol